src/Native/Utils.js

```diff
@@ -16,7 +16,13 @@ function eq(rootX, rootY)
 		{
 			continue;
 		}
-		if (typeof x === 'object')
+
+		if (x instanceof Date && y instanceof Date)
+		{
+			if (x.valueOf() !== y.valueOf())
+			{ return false; }
```

Comment:
Evan's preferred format is to stick the return on the next line too, then align the closing bracket with the first